gpii/node_modules/orca/orcaSettingsHandler.js

```diff
@@ -53,129 +41,98 @@
 
-    // TODO: Remove this method and all the blocking stuff.
-    // (see https://github.com/GPII/linux/pull/16)
-    //
-    // This setting handler has to be converted to an asynchronous as soon as
-    // the lifecycle manager can handle asynchronous actions.
-    //
-    // The reason for this blocking code is because we need a mechanism to know
-    // when the Orca's settings file has been created.
-    //
-    // The blocking "while loop" is where this method is used and is a big
-    // stability risk to the system, since it blocks the node event loop
-    // possibly indefinitely.
-    //
-    // Once we can make this setting handler to be asynchronous, we can use
-    // fs.watch or any other similar asynchronous approaches to know
-    // when the orca settings file is being created.
-    //
-    function wait(millis) {
-        var date = new Date();
-        var curDate;
-
-        do { curDate = new Date(); }
-        while (curDate - date < millis);
-    }
-
-    function applySettings(app) {
-        var settings = app.settings;
-
-        var userSettings = require(orcaSettingsFile);
-        var currentSettingsInProfile = userSettings.profiles[userSettings.general.activeProfile[PROFILE_ID]];
-
-        var applier = fluid.makeChangeApplier(currentSettingsInProfile, {
+    fluid.defaults("gpii.orca.settingsHandler", {
+        gradeNames: ["fluid.modelComponent"],
+        changeApplierOptions: {
             resolverGetConfig: fluid.model.escapedGetConfig,
             resolverSetConfig: fluid.model.escapedSetConfig
-        });
-
-        var newSettingsResponse = {};
-        fluid.each(settings, function (settingVal, settingKey) {
-            var oldValue = fluid.get(currentSettingsInProfile, settingKey,
-                fluid.model.escapedGetConfig);
+        },
+        model: {
+            settings: {}
+        },
+        invokers: {
+            get: {
+                funcName: "gpii.orca.settingsHandler.get",
+                args: ["{that}", "{arguments}.0"]
+            },
+            set: {
+                funcName: "gpii.orca.settingsHandler.set",
+                args: ["{that}", "{arguments}.0"]
+            },
+            getImpl: {
+                funcName: "gpii.orca.settingsHandler.getImpl",
+                args: ["{that}", "{arguments}.0"]
+            },
+            setImpl: {
+                funcName: "gpii.orca.settingsHandler.setImpl",
+                args: ["{that}", "{arguments}.0"]
+            }
+        },
+        listeners: {
+            onCreate: {
+                funcName: "gpii.orca.settingsHandler.onCreate",
+                args: ["{that}"]
+            }
+        }
+    });
 
-            var change = {
-                path: settingKey,
-                value: settingVal
-            };
-            change.type = settingVal === undefined ? "DELETE" : "ADD";
-            applier.fireChangeRequest(change);
+    gpii.orca.settingsHandler.onCreate = function (that) {
```

Comment:
@amb26, my only question is,

how about the get and set methods? Will they catch the rejected promise when they are trying to resolve model.settings and throw the corresponding error, or do I need to add something else?